File: src/lib/api/directoryClient.ts
import * as pb from "../pb";
import { RPCHost } from "../rpc/host";
import { Readable as GenericReadable } from "../rpc/stream";

export default class Directory {
  constructor(private readonly host: RPCHost) {}

  public publish(
    arg: pb.IDirectoryPublishRequest = new pb.DirectoryPublishRequest()
  ): Promise<pb.DirectoryPublishResponse> {
    return this.host.expectOne(
      this.host.call("Directory/Publish", new pb.DirectoryPublishRequest(arg))
    );
  }
  public unpublish(
    arg: pb.IDirectoryUnpublishRequest = new pb.DirectoryUnpublishRequest()
  ): Promise<pb.DirectoryUnpublishResponse> {
    return this.host.expectOne(
      this.host.call("Directory/Unpublish", new pb.DirectoryUnpublishRequest(arg))
    );
  }
  public join(
    arg: pb.IDirectoryJoinRequest = new pb.DirectoryJoinRequest()
  ): Promise<pb.DirectoryJoinResponse> {
    return this.host.expectOne(this.host.call("Directory/Join", new pb.DirectoryJoinRequest(arg)));
  }
  public part(
    arg: pb.IDirectoryPartRequest = new pb.DirectoryPartRequest()
  ): Promise<pb.DirectoryPartResponse> {
    return this.host.expectOne(this.host.call("Directory/Part", new pb.DirectoryPartRequest(arg)));
  }
  public ping(
    arg: pb.IDirectoryPingRequest = new pb.DirectoryPingRequest()
  ): Promise<pb.DirectoryPingResponse> {
    return this.host.expectOne(this.host.call("Directory/Ping", new pb.DirectoryPingRequest(arg)));
  }
}

File: android/app/src/main/java/gg/strims/ppspp/rpc/DirectoryClient.kt
package gg.strims.ppspp.rpc
import gg.strims.ppspp.proto.*

class DirectoryClient(filepath: String) : RPCClient(filepath) {

    suspend fun publish(
        arg: DirectoryPublishRequest = DirectoryPublishRequest()
    ): DirectoryPublishResponse =
        this.callUnary("Directory/Publish", arg)

    suspend fun unpublish(
        arg: DirectoryUnpublishRequest = DirectoryUnpublishRequest()
    ): DirectoryUnpublishResponse =
        this.callUnary("Directory/Unpublish", arg)

    suspend fun join(
        arg: DirectoryJoinRequest = DirectoryJoinRequest()
    ): DirectoryJoinResponse =
        this.callUnary("Directory/Join", arg)

    suspend fun part(
        arg: DirectoryPartRequest = DirectoryPartRequest()
    ): DirectoryPartResponse =
        this.callUnary("Directory/Part", arg)

    suspend fun ping(
        arg: DirectoryPingRequest = DirectoryPingRequest()
    ): DirectoryPingResponse =
        this.callUnary("Directory/Ping", arg)

}
